### DISCUSSION TITLE
Identifying individual health events

### DISCUSSION THREAD

**Discussion by SRW612** [opensafely/documentation#262] [Q&A] [answered]
Our study requires counting the number of events satisfying a given codelist (e.g. codelist for RTIs). For example, if patient X attends their GP on two different occasions during the study period, and are coded as having an RTI both times, we need to count both events for that patient. However, the study_definition.py template is set up for patient-level data. Is there a way to derive counts of events rather than patients by either
1. defining the study population at event-level, where the two events for Patient X would be represented by two rows.
OR
2. identifing multiple events for a given patient from patient-level data, where Patient X is represented by one row, but the two events for that patient are represented by other variables in the row, e.g. Event1='CTV3 code for RTI', Event2='CTV3 code for RTI' ?

**Comment by alexwalkerepi** [comment, marked as answer]
Currently there isn't a straightforward way to return all the individual events themselves. We hope to add this functionality eventually.

However, if you only need to _count_ the individual events, you can return the count of recorded events within a given time period by using the `returning="number_of_matches_in_period"` option, which will return an integer count, as documented here:
https://docs.opensafely.org/study-def-variables/#cohortextractor.patients.with_these_clinical_events

Does that fulfil your use case?

**Reply by SRW612** [reply]
Many thanks Alex, I think the _with_gp_consultations_ function will work for our needs.

**Reply by SRW612** [reply]
Just to clarify, can this approach be used to count consultations defined by a particular diagnosis, e.g. only consultations with a respiratory infection diagnosis? If not then I assume I have to look into Will's suggtions below?

**Reply by alexwalkerepi** [reply]
For that you'll need the `patients.with_these_clinical_events` function, which you use with a codelist to define the condition you're interested in. That also has the `returning="number_of_matches_in_period"` option.

**Reply by SRW612** [reply]
Brilliant, I think that will do it.

**Comment by wjchulme** [comment]
Option 1 is not currently possible -- see [this issue](https://github.com/opensafely-core/cohort-extractor/issues/136) and the issues linked therein for more info about how we're working to resolve this.

Option 2 is possible with a bit of guile. See section *Extracting a series of consecutive events* in [this issue](https://github.com/opensafely/documentation/issues/252). **WARNING** I wouldn't recommend this approach if you want to return a large number of events. This would mean you need to create lots of variables (columns) which means more queries and slower extraction time. It might be useful for you to talk through your use-case in a bit more detail first.

**Reply by SRW612** [reply]
Thanks for the alternative appraoch. If the _with_gp_consultations_ function isn't right for our study I will try your suggestions for option 2.
